### PR TITLE
Autoplay

### PR DESCRIPTION
**Pull request by misteroneill**

### CODE REVIEW

**Comment by gkatsev** [comment]
Deploy preview for *videojs-blog* ready!

Built with commit 07aa246818b8b525b67007e8c81ffb48fe2f3483

https://deploy-preview-28--videojs-blog.netlify.com

File: source/_drafts/Autoplay-Best-Practices-with-Video-js.md [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+## Autoplay in the Big Browsers
+I'm not going to go into exhaustive detail on each browser's idiosyncrasies because that defeats the point: **never assume autoplay will work.**
+
+### Chrome
+Back in September 2017, Google announced that [Chrome's autoplay policy would change in April 2018 with Chrome 66](https://developers.google.com/web/updates/2017/09/autoplay-policy-changes). In short, muted autoplay is allowed, but autoplay with sound will be subject to a series of rules depending on platform and previous video engagement (referred to as the "Media Engagement Index") with the specific domain.
+
+### Safari
+As of Safari 11, which shipped in September 2017, Apple has [updated their autoplay policies](https://webkit.org/blog/7734/auto-play-policy-changes-for-macos/) to prevent autoplay with sound on most websites.
+
+### Firefox
```
Comment (gkatsev) [inline source/_drafts/Autoplay-Best-Practices-with-Video-js.md]: Firefox's approach is basically wait and see where things fall. The option that's available has been around for a long time. Probably still good to mention it, though.

File: source/_drafts/Autoplay-Best-Practices-with-Video-js.md [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+   <video-js autoplay src="https://path/to/source.mp4"></video-js>
+   ```
+
+1. Use the `autoplay` option:
+
+   ```js
+   videojs('player', {autoplay: true});
+   ```
+
+## Recommended Practice
```
Comment (gkatsev) [inline source/_drafts/Autoplay-Best-Practices-with-Video-js.md]: would be good to mention can-autoplay project.

File: source/_drafts/Autoplay-Best-Practices-with-Video-js.md [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+
+This post is not about ways to circumvent autoplay policies. That's not possible and no one should attempt it. One positional statement everyone working on Video.js should be comfortable making is: actively circumventing the browser's built-in user experience or user preferences is harmful.
+
+## Autoplay in the Big Browsers
+I'm not going to go into exhaustive detail on each browser's idiosyncrasies because that defeats the point: **never assume autoplay will work.**
+
+### Chrome
+Back in September 2017, Google announced that [Chrome's autoplay policy would change in April 2018 with Chrome 66](https://developers.google.com/web/updates/2017/09/autoplay-policy-changes). In short, muted autoplay is allowed, but autoplay with sound will be subject to a series of rules depending on platform and previous video engagement (referred to as the "Media Engagement Index") with the specific domain.
+
+### Safari
```
Comment (gkatsev) [inline source/_drafts/Autoplay-Best-Practices-with-Video-js.md]: should safari go first since they were first in implementing this?

File: source/_drafts/Autoplay-Best-Practices-with-Video-js.md [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+
+This post is not about ways to circumvent autoplay policies. That's not possible and no one should attempt it. One positional statement everyone working on Video.js should be comfortable making is: actively circumventing the browser's built-in user experience or user preferences is harmful.
+
+## Autoplay in the Big Browsers
+I'm not going to go into exhaustive detail on each browser's idiosyncrasies because that defeats the point: **never assume autoplay will work.**
+
+### Chrome
+Back in September 2017, Google announced that [Chrome's autoplay policy would change in April 2018 with Chrome 66](https://developers.google.com/web/updates/2017/09/autoplay-policy-changes). In short, muted autoplay is allowed, but autoplay with sound will be subject to a series of rules depending on platform and previous video engagement (referred to as the "Media Engagement Index") with the specific domain.
```
Comment (gkatsev) [inline source/_drafts/Autoplay-Best-Practices-with-Video-js.md]: should we mention the gesture thing?

File: source/_drafts/Autoplay-Best-Practices-with-Video-js.md (hunk 1 above)
Comment (gkatsev) [reply source/_drafts/Autoplay-Best-Practices-with-Video-js.md]: might be good to say that they may implement an autoplay policy in the future.

File: source/_drafts/Autoplay-Best-Practices-with-Video-js.md (hunk 3 above)
Comment (gkatsev) [inline source/_drafts/Autoplay-Best-Practices-with-Video-js.md]: should we mention that once the user unmuted a video, all subsequent videos will play?

File: source/_drafts/Autoplay-Best-Practices-with-Video-js.md [hunk 5]
```diff
@@ -0,0 +1,113 @@
+---
+title: "Autoplay Best Practices with Video.js"
+author:
+  name: "Pat O'Neill"
+  github: misteroneill
+tags:
+---
+
```
Comment (gkatsev) [inline source/_drafts/Autoplay-Best-Practices-with-Video-js.md:8]: Thoughts about having a `tl;dr` section that basically says "use player.play() for autoplaying" before the post?

File: source/_drafts/Autoplay-Best-Practices-with-Video-js.md (hunk 3 above)
Comment (misteroneill) [reply source/_drafts/Autoplay-Best-Practices-with-Video-js.md]: I went back and forth on this. The reason for the post is that Chrome 66 is coming, so I put it first, but I think maybe Safari should go first here.

File: source/_drafts/Autoplay-Best-Practices-with-Video-js.md (hunk 4 above)
Comment (misteroneill) [reply source/_drafts/Autoplay-Best-Practices-with-Video-js.md]: I was intentionally trying not to explain each browser's particular algorithm because that's more likely to change over time and people can read about it from the browser vendors themselves, if they care.

File: source/_drafts/Autoplay-Best-Practices-with-Video-js.md (hunk 5 above)
Comment (misteroneill) [reply source/_drafts/Autoplay-Best-Practices-with-Video-js.md:8]: Sounds good.

File: source/_drafts/Autoplay-Best-Practices-with-Video-js.md (hunk 4 above)
Comment (gkatsev) [reply source/_drafts/Autoplay-Best-Practices-with-Video-js.md]: sounds good.

File: source/_drafts/Autoplay-Best-Practices-with-Video-js.md [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+As of Safari 11, which shipped in September 2017, Apple has [updated their autoplay policies](https://webkit.org/blog/7734/auto-play-policy-changes-for-macos/) to prevent autoplay with sound on most websites.
+
+### Chrome
+Back in September 2017, Google announced that [Chrome's autoplay policy would change in April 2018 with Chrome 66](https://developers.google.com/web/updates/2017/09/autoplay-policy-changes), subject to a series of rules you can read about in the linked article.
+
+### Firefox
+With Firefox, Mozilla has taken the position of not having a firm autoplay policy for the time being. That may change in the future. That said, Firefox today [does allow users to disable autoplay](https://support.mozilla.org/en-US/questions/1150702) with a few configuration changes.
+
+### IE11 and Edge
+Autoplay always works in Microsoft's browsers.
```
Comment (ldayananda) [inline source/_drafts/Autoplay-Best-Practices-with-Video-js.md]: I guess rephrase this to say that IE11 and Edge do not have specific autoplay restrictions currently? Presumably Edge might decide to introduce an autoplay policy later on

**Review by ldayananda — APPROVED** [review]
one comment, otherwise LGTM

File: source/_drafts/Autoplay-Best-Practices-with-Video-js.md (hunk 6 above)
Comment (misteroneill) [reply source/_drafts/Autoplay-Best-Practices-with-Video-js.md]: Fair enough!

**Review by kcj382 — APPROVED** [review]

**Review by brandonocasey — APPROVED** [review]
